lib/GADS/Role/Purgable.pm

```diff
@@ -0,0 +1,29 @@
+package GADS::Role::Purgable;
+
+use strict;
+use warnings;
+
+use Moo::Role;
+
+has recordsource => (
+    is      => 'lazy',
+    builder => sub { undef; }
+);
```

Comment:
There should be no need for `recordsource` anymore. The only place it is used (`GADS::Schema::Result::File`) can obtain it from itself.